### DISCUSSION TITLE
ResourceWarning: unclosed file

### DISCUSSION THREAD

**Discussion by erkib** [encode/django-rest-framework#8998] [Question & Answer]
Python: 3.8.16
Django: 4.2.1
DRF: 3.14.0

When using ModelSerializer with FileField on Model and doing file upload with DRF, in the end of tests it shows:

...
OK
Destroying test database for alias 'default'...
sys:1: ResourceWarning: unclosed file <_io.BufferedReader name='/Users/xxx/SourceCode/private/xxx/_local/mediafiles/user_1/profile/xxx.png'>
ResourceWarning: Enable tracemalloc to get the object allocation traceback

**Comment by jos-h** [comment]
@erkib Can you check in your code that the file has been closed after you have finished with upload. This warning is coming from the Cpython code. See [here](https://github.com/python/cpython/blob/d593074494b39a3d51e67a4e57189c530867a7ff/Doc/library/devmode.rst?plain=1#L142)

**Comment by erkib** [comment]
I'm using the `with`-pattern as suggested by the django documentation and passing the file handle to upload model.
